### PR TITLE
reserved names prevention on entity name or field

### LINKED ISSUE #2269: 🐛 Bug Report: Generated app fails to build when reserved names are used for an entity name
### What happened?

When a user creates an entity with a reserved name, the generated app failed to build.

For example, when the user uses the name "Error" for an entity, the following exception is thrown at Cloud Build when building the docker image for the sandbox

```
Step #1: Step #2: /app/server/node_modules/ts-node/src/index.ts:513
Step #1: Step #2:     return new TSError(diagnosticText, diagnosticCodes)
Step #1: Step #2:            ^
Step #1: Step #2: TSError: ⨯ Unable to compile TypeScript:
Step #1: Step #2: scripts/seed.ts(14,21): error TS2554: Expected 0 arguments, but got 1.
```

The same error can see seen when running the seed script locally

image

### What you expected to happen

The user should not be able to create entities (and entity fields) with names that are reserved

### How to reproduce

Create an entity with the name Error and commit the changes to deploy the code to sandbox - the deployment will fail

### Amplication version

0.11.2

### Environment

_No response_

### Are you willing to submit PR?

_No response_

### PR DESCRIPTION
**Pull request by noyjs**
should fix #2269

### CODE REVIEW

File: packages/amplication-server/src/core/entity/entity.service.ts [hunk 1]
```diff
@@ -234,6 +237,13 @@ export class EntityService {
         `The entity name and plural display name cannot be the same.`
       );
     }
+    if (isReservedName(args.data?.name?.toLowerCase().trim())) {
+      throw new AmplicationError(
+        `"${args.data?.name
+          ?.toLowerCase()
+          .trim()}" is a reserved name and cannot be used.`
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/entity/entity.service.ts:244]: Create a new Error and use a string template in the contractor to generate the message
see this example 
packages/amplication-server/src/core/app/ReservedEntityNameError.ts

File: packages/amplication-server/src/core/entity/entity.service.ts [hunk 2]
```diff
@@ -125,6 +126,8 @@ export const NAME_VALIDATION_ERROR_MESSAGE =
 
 export const DELETE_ONE_USER_ENTITY_ERROR_MESSAGE = `The 'user' entity is a reserved entity and it cannot be deleted`;
 
+export const CANNOT_USE_RESERVED_NAME_ERROR_MESSAGE = `"class" is a reserved name and cannot be used.`;
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/entity/entity.service.ts:129]: Remove this const

File: packages/amplication-server/src/core/entity/entity.service.ts [hunk 3]
```diff
@@ -567,6 +577,14 @@ export class EntityService {
         );
       }
 
+      if (isReservedName(args.data?.name?.toLowerCase().trim())) {
+        throw new AmplicationError(
+          `"${args.data?.name
+            ?.toLowerCase()
+            .trim()}" is a reserved name and cannot be used.`
+        );
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/entity/entity.service.ts:585]: same as above (use custom Error)

File: packages/amplication-server/src/core/entity/entity.service.ts [hunk 4]
```diff
@@ -1812,6 +1830,14 @@ export class EntityService {
     args: CreateOneEntityFieldArgs,
     user: User
   ): Promise<EntityField> {
+    if (isReservedName(args.data?.name?.toLowerCase().trim())) {
+      throw new AmplicationError(
+        `"${args.data?.name
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/entity/entity.service.ts:1835]: same as above (use custom Error)

File: packages/amplication-server/src/core/entity/entity.service.ts [hunk 5]
```diff
@@ -2015,6 +2041,13 @@ export class EntityService {
     args: UpdateOneEntityFieldArgs,
     user: User
   ): Promise<EntityField> {
+    if (isReservedName(args.data?.name?.toLowerCase().trim())) {
+      throw new AmplicationError(
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/entity/entity.service.ts:2045]: same as above (use custom Error)

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
well done
some minor requests for changes

**Review by yuval-hazaz — APPROVED** [review]

**Comment by yuval-hazaz** [comment]
@mshidlov please also review this PR

**Comment by mshidlov** [comment]
@noyjs thank you for submitting your PR, I finished doing some tests and this looks good. I'm going to merge this PR to release `0.11.3`

**Pull request merged**